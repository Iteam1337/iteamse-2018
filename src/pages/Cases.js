// @flow

import * as React from 'react'
import * as Iteam from '../typings/iteam.flow'
import * as IteamCMS from './__generated__/CasesPage'
import GridColumn from '../components/Grid/GridColumn'
import PaddedRow from '../components/Grid/PaddedRow'
import styled from 'styled-components'
import Team from '../components/Team/Team'
import Header from '../components/Header/Header'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import { CasePageQuery } from './Case'
import PrefetchLink from '../components/Link/PrefetchLink'
import { media } from '../theme'

type QueryProps = Iteam.ApolloBase<IteamCMS.CasesPage>

export const CasesPageQuery = gql`
  query CasesPage {
    pageCases {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
    }
    cases {
      location
      shortDescription
      slug
      thumbnailImage
      title
    }
  }
`

const Cases = PaddedRow.extend`
  display: grid;
  grid-row-gap: 40px;
  grid-template-columns: 1fr;

  ${media.desktop`
    grid-column-gap: 30px;
    grid-template-columns: repeat(2, 1fr);
  `};
`

const CaseImageWrap = styled.div`
  align-items: center;
  background-color: #f1f1f1;
  display: flex;
  justify-content: center;

  ${media.desktop`
    height: 500px;
    width: 500px;
  `};
`

const CaseImage = styled.img`
  max-width: 90%;
`

const Meta = styled.div`
  border-left: 0px solid ${({ theme }) => theme.colors.aquamarine};
  margin-top: 30px;
  transition-property: border-left-width, padding-left;
  transition-duration: 300ms;
  transition-timing-function: ease-in-out;
`

const CaseLink = styled(PrefetchLink)`
  color: #000;
  text-decoration: none;
`

const Case = styled.div`
  &:hover {
    ${Meta} {
      border-left-width: 10px;
      padding-left: 10px;
    }
  }
`

const Title = styled.div`
  font-size: 18px;
  font-weight: 500;
  margin-bottom: 10px;
`

const ShortDescription = styled.div`
  font-size: 25px;
  font-weight: 300;
`

export const CasePage = () => {
  return (
    <Query query={CasesPageQuery}>
      {({ loading, data: { pageCases, cases } }: QueryProps) => {
        if (loading) {
          return null
        }

        return (
          <React.Fragment>
            <Header
              backgroundImage={pageCases.headerImage}
              messageBgColor={pageCases.headerTextBgColor}
              messageOne={pageCases.headerText1}
              messageTwo={pageCases.headerText2}
            />

            <GridColumn>
              <Cases>
                {cases.map(workCase => (
                  <CaseLink
                    query={CasePageQuery}
                    key={workCase.title}
                    to={`/case/${workCase.slug}`}
                    variables={{
                      slug: workCase.slug,
                    }}
                  >
                    <Case>
                      <CaseImageWrap>
                        {workCase.thumbnailImage && (
                          <CaseImage src={workCase.thumbnailImage} alt="" />
                        )}
                      </CaseImageWrap>
                      <Meta>
                        <Title>{workCase.title}</Title>
                        <ShortDescription>
                          {workCase.shortDescription}
                        </ShortDescription>
                      </Meta>
                    </Case>
                  </CaseLink>
                ))}
              </Cases>
            </GridColumn>

            <Team bgColor="green" shortName={['hrn', 'jmn']} />
          </React.Fragment>
        )
      }}
    </Query>
  )
}

export default CasePage
