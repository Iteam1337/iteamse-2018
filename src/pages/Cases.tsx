import gql from 'graphql-tag'
import * as React from 'react'
import { Query } from 'react-apollo'
import { Helmet } from 'react-helmet'
import { CasesPageQuery } from '../../typings/iteamse'
import GridColumn from '../components/Grid/GridColumn'
import PaddedRow from '../components/Grid/PaddedRow'
import Header from '../components/Header/Header'
import PrefetchLink from '../components/Link/PrefetchLink'
import Team from '../components/Team/Team'
import styled from '../theme'
import { CASE_PAGE_QUERY } from './Case'

export const CASES_PAGE_QUERY = gql`
  query CasesPage {
    pageCases {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
      contactTitle
      team
    }
    cases {
      location
      shortDescription
      slug
      thumbnailImage
      title
    }
  }
`

const Cases = styled(PaddedRow)`
  display: grid;
  grid-row-gap: 40px;
  grid-template-columns: 1fr;

  @media (min-width: 1025px) {
    grid-column-gap: 30px;
    grid-template-columns: repeat(2, 1fr);
  }

  /* IE 11 */
  ${({ theme }) =>
    theme.browsers.ie10Or11(`
    width: 100%;
    display: flex;
    align-items: flex-start;
    justify-content: center;
    flex-wrap: wrap;

    > * {
      flex: 1 0 50%;
      box-sizing: border-box;
    }

    > a {
      display: block;
      margin-bottom: 25px;
    }
  `)};
`

const CaseImageWrap = styled.div`
  align-items: center;
  background-color: #f1f1f1;
  display: flex;
  justify-content: center;

  @media (min-width: 1025px) {
    height: 500px;
    width: 500px;
  }
`

const CaseImage = styled.img`
  max-width: 90%;

  /* IE 11 */
  ${({ theme }) =>
    theme.browsers.ie10Or11(`
    max-width: 396px;
    border: none;
  `)};
`

const Meta = styled.div`
  border-left: 0px solid ${({ theme }) => theme.colors.aquamarine};
  margin-top: 30px;
  transition-property: border-left-width, padding-left;
  transition-duration: 300ms;
  transition-timing-function: ease-in-out;
`

const CaseLink = styled(PrefetchLink)`
  color: #000;
  text-decoration: none;
`

const Case = styled.div`
  &:hover {
    ${Meta} {
      border-left-width: 10px;
      padding-left: 10px;
    }
  }

  /* IE 11 */
  ${({ theme }) =>
    theme.browsers.ie10Or11(`
    max-width: 496px;
  `)};
`

const Title = styled.div`
  font-size: 18px;
  font-weight: 500;
  margin-bottom: 10px;
`

const ShortDescription = styled.div`
  font-size: 25px;
  font-weight: 300;
`

class CasesQuery extends Query<CasesPageQuery> {}

export class CasePage extends React.Component {
  render() {
    return (
      <CasesQuery query={CASES_PAGE_QUERY}>
        {({ loading, data }) => {
          if (loading || !data) {
            return null
          }

          const { pageCases, cases } = data

          return (
            <>
              <Helmet>
                <title>Iteam |Â Case</title>
                <meta property="og:title" content="Iteam | Case" />
                <meta property="twitter:title" content="Iteam | Case" />
                {pageCases.headerImage && (
                  <meta
                    property="og:image"
                    content={`https:${pageCases.headerImage}`}
                  />
                )}
              </Helmet>
              <Header
                backgroundImage={pageCases.headerImage}
                messageBgColor={pageCases.headerTextBgColor}
                messageOne={pageCases.headerText1}
                messageTwo={pageCases.headerText2}
              />

              <GridColumn>
                <Cases>
                  {cases.map(workCase => {
                    if (!workCase) {
                      return null
                    }

                    return (
                      <CaseLink
                        query={CASE_PAGE_QUERY}
                        key={workCase.title}
                        to={`/case/${workCase.slug}`}
                        variables={{
                          slug: workCase.slug,
                        }}
                      >
                        <Case>
                          <CaseImageWrap>
                            {workCase.thumbnailImage && (
                              <CaseImage src={workCase.thumbnailImage} alt="" />
                            )}
                          </CaseImageWrap>
                          <Meta>
                            <Title>{workCase.title}</Title>
                            <ShortDescription>
                              {workCase.shortDescription}
                            </ShortDescription>
                          </Meta>
                        </Case>
                      </CaseLink>
                    )
                  })}
                </Cases>
              </GridColumn>

              <Team
                bgColor="green"
                callToAction={pageCases.contactTitle}
                shortName={pageCases.team}
              />
            </>
          )
        }}
      </CasesQuery>
    )
  }
}

export default CasePage
