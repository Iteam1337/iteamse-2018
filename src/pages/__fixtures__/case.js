export const workCase = {
  headerBgColor: '#1616B2',
  logo:
    '//images.ctfassets.net/rj4r6yfcesw5/4p0ptdLj9SaECekOI4c6C2/c2cdeab1cebc9650ebb0e7778357c231/AF_ordbild_vit.png',
  slug: 'arbetsformedlingen',
  thumbnailImage:
    '//images.ctfassets.net/rj4r6yfcesw5/3Y7zZ25WbeOCU8w84kAEos/b97ff1899cf6f2e4dca0eb2b4f0d5a44/af.png',
  tags: ['API', 'Docker', 'UX', 'React', 'Vue'],
  title: 'Arbetsförmedlingen',
  introduction:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  introductionTitle: 'En ny digital tjänst',
  process:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  processTitle: 'Process',
  development:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  developmentTitle: 'Utveckling',
  aboutCompany:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  aboutCompanyTitle: 'Arbetsförmedlingen',
  partners: '[Digitalbyrån](https://hej.com)',
  partnersTitle: 'Samarbetspartners',
  contact:
    '- [Hans Rollman](mailto:hans.rollman@iteam.se)\n- Johan Littorin, Telenor, 0734 41 72 85',
  contactTitle: 'Presskontakt',
  quote:
    '"We feel that our crew at Iteam are more colleagues than suppliers. Rather than specifying a tech spec to work on, we collaborate, carefully chiseling out functionality as we go along. This way of working has minimized waste and it has also taken us further than we ever imagined”',
  quotePerson: 'Anna Andersson, chef Arbetsförmedlingen',
  __typename: 'Case',
}
