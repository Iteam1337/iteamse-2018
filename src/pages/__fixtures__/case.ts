export const workCase = {
  __typename: 'Case',
  aboutCompany:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  aboutCompanyTitle: 'Arbetsförmedlingen',
  contact:
    '- [Hans Rollman](mailto:hans.rollman@iteam.se)\n- Johan Littorin, Telenor, 0734 41 72 85',
  contactTitle: 'Presskontakt',
  development:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  developmentTitle: 'Utveckling',
  headerBgColor: '#1616B2',
  introduction:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  introductionTitle: 'En ny digital tjänst',
  logo:
    '//images.ctfassets.net/rj4r6yfcesw5/4p0ptdLj9SaECekOI4c6C2/c2cdeab1cebc9650ebb0e7778357c231/AF_ordbild_vit.png',
  partners: '[Digitalbyrån](https://hej.com)',
  partnersTitle: 'Samarbetspartners',
  process:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  processTitle: 'Process',
  quote:
    '"We feel that our crew at Iteam are more colleagues than suppliers. Rather than specifying a tech spec to work on, we collaborate, carefully chiseling out functionality as we go along. This way of working has minimized waste and it has also taken us further than we ever imagined”',
  quotePerson: 'Anna Andersson, chef Arbetsförmedlingen',
  slug: 'arbetsformedlingen',
  tags: ['API', 'Docker', 'UX', 'React', 'Vue'],
  team: null,
  thumbnailImage:
    '//images.ctfassets.net/rj4r6yfcesw5/3Y7zZ25WbeOCU8w84kAEos/b97ff1899cf6f2e4dca0eb2b4f0d5a44/af.png',
  title: 'Arbetsförmedlingen',
}
