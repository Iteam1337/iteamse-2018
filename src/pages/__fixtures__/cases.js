export const cases = [
  {
    location: 'Stockholm',
    shortDescription:
      'Synliggörande av asylsökandes kompetens med digitala verktyg',
    slug: 'vimla',
    thumbnailImage:
      '//images.ctfassets.net/rj4r6yfcesw5/68OlBclfQAGMcEUsQgcIUY/9bdf52bb8a9a17be06e11ef3fa32d3de/vimla-case-phone.png',
    title: 'Vimla',
    __typename: 'Case',
  },
  {
    location: 'Stockholm',
    shortDescription:
      'Synliggörande av asylsökandes kompetens med digitala verktyg',
    slug: 'tsab',
    thumbnailImage:
      '//images.ctfassets.net/rj4r6yfcesw5/UGucAeFycSqIGw0CAc0ys/3d58300261bcff64dba751de9228a150/tsab-case-phone.png',
    title: 'Taxi Stockholm',
    __typename: 'Case',
  },
  {
    location: 'Stockholm',
    shortDescription:
      'Synliggörande av asylsökandes kompetens med digitala verktyg',
    slug: 'seb',
    thumbnailImage: null,
    title: 'SEB',
    __typename: 'Case',
  },
  {
    location: 'Stockholm',
    shortDescription:
      'Synliggörande av asylsökandes kompetens med digitala verktyg',
    slug: 'arbetsformedlingen',
    thumbnailImage:
      '//images.ctfassets.net/rj4r6yfcesw5/3Y7zZ25WbeOCU8w84kAEos/b97ff1899cf6f2e4dca0eb2b4f0d5a44/af.png',
    title: 'Arbetsförmedlingen',
    __typename: 'Case',
  },
]
