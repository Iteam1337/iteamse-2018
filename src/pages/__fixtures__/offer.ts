// tslint:disable:object-literal-sort-keys
export const offers = [
  {
    offerIntroImage:
      '//images.ctfassets.net/rj4r6yfcesw5/5n7J4mzx3UWam0YwgsUQm/8e00d12d6e80eba927b2a313fb250d87/1.png',
    offerTitle: 'Kod',
    offerIntroText:
      'Systemutveckling, automatisering, artificiell intelligens, maskininlärning. Vi har alla redskap du behöver för att bygga vad du vill. Det du vill. Och vet du inte vad du behöver så hjälper vi dig med det med. ',
    offerLead:
      'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. ',
    offerUspOneTitle: 'Utveckling av pilot',
    offerUspOneText:
      'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi.',
    offerUspTwoTitle: 'API',
    offerUspTwoText:
      'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi.',
    offerUspThreeTitle: 'Mobila applikationer',
    offerUspThreeText:
      'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi.',
    offerUspFourTitle: 'AI-lösningar',
    offerUspFourText:
      'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi\n',
    offerIllustrationImage:
      '//images.ctfassets.net/rj4r6yfcesw5/5SAhc8YuL6OaQQsC6miokK/7c09d594e44ffa47781bf1bdb21aae7f/illustrations-group-2.png',
    offerOrder: 1,
    __typename: 'Offer',
  },
]
