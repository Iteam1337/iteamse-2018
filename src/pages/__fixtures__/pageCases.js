export const pageCases = {
  headerImage:
    '//images.ctfassets.net/rj4r6yfcesw5/45ST2zZeFqgYuaSuM8yKy0/455471362d58784653089b0f77b2c4e9/Maria-Desk-2-Highres.jpg',
  headerText1: 'Några exempel på',
  headerText2: 'sådant vi gjort',
  headerTextBgColor: 'green',
  __typename: 'PageCases',
}
