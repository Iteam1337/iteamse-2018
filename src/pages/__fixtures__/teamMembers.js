export const teamMembers = [
  {
    avatar: '',
    email: 'cm@sesame.com',
    gravatar: 'cm.png',
    location: 'Stockholm',
    name: 'Cookie M',
    phoneNumber: '0701234567',
    title: 'Cookie master',
    short: 'cmm',
    __typename: 'TeamMember',
  },
  {
    avatar: '',
    email: 'count@sesame.com',
    gravatar: 'count.png',
    location: 'GÃ¶teborg',
    name: 'Count van Count',
    phoneNumber: '0707654321',
    title: '1 ahahah',
    short: 'cvc',
    __typename: 'TeamMember',
  },
]
