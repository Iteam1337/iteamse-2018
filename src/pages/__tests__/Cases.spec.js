import React from 'react'
import { CasePage } from '../Cases'
import { render, wait } from 'react-testing-library'
import MockedQuery from '../../utils/test-utils/MockedQuery'
import { pageCases } from '../__fixtures__/pageCases'
import { teamMembers } from '../__fixtures__/teamMembers'
import { cases } from '../__fixtures__/cases'

describe('components/Cases', () => {
  const mockedResponse = {
    cases,
    pageCases,
    teamMembers,
  }

  it('renders Cases', async () => {
    const { container } = render(
      <MockedQuery response={mockedResponse}>
        <CasePage />
      </MockedQuery>
    )

    await wait()

    expect(container).toMatchSnapshot()
  })
})
