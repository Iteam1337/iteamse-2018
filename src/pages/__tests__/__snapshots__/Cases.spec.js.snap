// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Cases renders Cases 1`] = `
<div>
  <div
    style="height: 700px;"
    class="lazyload-placeholder"
  />
  <div
    class="s7kn9a1-0-GridColumn__GridColumnClean-lcQAdD iyGAEe"
  >
    <div
      class="s3nsc5b-0-PaddedRow-hWTqUj iUATLn"
    >
      <div
        class="Filter__Wrap-s16ccbkk-0 evelsV"
        data-test="filter-"
        tabindex="0"
      >
        Alla
      </div>
      <div
        class="Filter__Wrap-s16ccbkk-0 lgKsfF"
        data-test="filter-Stockholm"
        tabindex="0"
      >
        Stockholm
      </div>
      <div
        class="Filter__Wrap-s16ccbkk-0 lgKsfF"
        data-test="filter-Göteborg"
        tabindex="0"
      >
        Göteborg
      </div>
    </div>
    <div
      class="s3nsc5b-0-PaddedRow-hWTqUj hbGnBu"
    >
      <a
        class="PrefetchLink__StyledLink-n3ecxs-0 JPiJu"
        href="/case/vimla"
      >
        <div
          class="Cases__Case-ql1p4z-1 iHJPWM"
        >
          <div
            class="Cases__CaseImageWrap-ql1p4z-2 bwMITc"
          >
            <img
              class="Cases__CaseImage-ql1p4z-3 bRuamq"
              src="//images.ctfassets.net/rj4r6yfcesw5/68OlBclfQAGMcEUsQgcIUY/9bdf52bb8a9a17be06e11ef3fa32d3de/vimla-case-phone.png"
              alt=""
            />
          </div>
          <div
            class="Cases__Meta-ql1p4z-4 fnbXh"
          >
            <div
              class="Cases__Title-ql1p4z-5 WtUOC"
            >
              Vimla
            </div>
            <div
              class="Cases__ShortDescription-ql1p4z-6 eAkhlI"
            >
              Synliggörande av asylsökandes kompetens med digitala verktyg
            </div>
          </div>
        </div>
      </a>
      <a
        class="PrefetchLink__StyledLink-n3ecxs-0 JPiJu"
        href="/case/tsab"
      >
        <div
          class="Cases__Case-ql1p4z-1 iHJPWM"
        >
          <div
            class="Cases__CaseImageWrap-ql1p4z-2 bwMITc"
          >
            <img
              class="Cases__CaseImage-ql1p4z-3 bRuamq"
              src="//images.ctfassets.net/rj4r6yfcesw5/UGucAeFycSqIGw0CAc0ys/3d58300261bcff64dba751de9228a150/tsab-case-phone.png"
              alt=""
            />
          </div>
          <div
            class="Cases__Meta-ql1p4z-4 fnbXh"
          >
            <div
              class="Cases__Title-ql1p4z-5 WtUOC"
            >
              Taxi Stockholm
            </div>
            <div
              class="Cases__ShortDescription-ql1p4z-6 eAkhlI"
            >
              Synliggörande av asylsökandes kompetens med digitala verktyg
            </div>
          </div>
        </div>
      </a>
      <a
        class="PrefetchLink__StyledLink-n3ecxs-0 JPiJu"
        href="/case/seb"
      >
        <div
          class="Cases__Case-ql1p4z-1 iHJPWM"
        >
          <div
            class="Cases__CaseImageWrap-ql1p4z-2 bwMITc"
          />
          <div
            class="Cases__Meta-ql1p4z-4 fnbXh"
          >
            <div
              class="Cases__Title-ql1p4z-5 WtUOC"
            >
              SEB
            </div>
            <div
              class="Cases__ShortDescription-ql1p4z-6 eAkhlI"
            >
              Synliggörande av asylsökandes kompetens med digitala verktyg
            </div>
          </div>
        </div>
      </a>
      <a
        class="PrefetchLink__StyledLink-n3ecxs-0 JPiJu"
        href="/case/arbetsformedlingen"
      >
        <div
          class="Cases__Case-ql1p4z-1 iHJPWM"
        >
          <div
            class="Cases__CaseImageWrap-ql1p4z-2 bwMITc"
          >
            <img
              class="Cases__CaseImage-ql1p4z-3 bRuamq"
              src="//images.ctfassets.net/rj4r6yfcesw5/3Y7zZ25WbeOCU8w84kAEos/b97ff1899cf6f2e4dca0eb2b4f0d5a44/af.png"
              alt=""
            />
          </div>
          <div
            class="Cases__Meta-ql1p4z-4 fnbXh"
          >
            <div
              class="Cases__Title-ql1p4z-5 WtUOC"
            >
              Arbetsförmedlingen
            </div>
            <div
              class="Cases__ShortDescription-ql1p4z-6 eAkhlI"
            >
              Synliggörande av asylsökandes kompetens med digitala verktyg
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    class="s7kn9a1-0-GridColumn__GridColumnClean-lcQAdD isotwF"
  >
    <div
      class="TeamMembers__Members-gavpkf-0 uMipK"
    >
      <div
        class="TeamMemberDuo__Colleague-s9kjazo-0 XxNhO"
      >
        <a
          class="PrefetchLink__StyledLink-n3ecxs-0 JPiJu"
          href="/teamet/cmm"
        >
          <div
            style="height: 226px;"
            class="lazyload-placeholder"
          />
        </a>
        <div
          class="TeamMemberDuo__Meta-s9kjazo-1 gzvJxw"
        >
          <div
            class="TeamMemberDuo__MetaSection-s9kjazo-3 XeOZD"
          >
            <div
              class="TeamMemberDuo__Title-s9kjazo-2 dFgPDl"
            >
              Cookie master
            </div>
            <div>
              Stockholm
            </div>
          </div>
          <div
            class="TeamMemberDuo__Title-s9kjazo-2 dFgPDl"
          >
            Cookie M
          </div>
          <div>
            <a
              class="s1qo6rwq-0-NativeLink-jXoiCT dJdJnO"
              href="tel:0701234567"
            >
              070-123 45 67
            </a>
          </div>
          <a
            class="NativeLink-s1qo6rwq-0 jNinQt"
            href="mailto:cm@sesame.com"
          >
            cm@sesame.com
          </a>
        </div>
      </div>
      <div
        class="TeamMemberDuo__Colleague-s9kjazo-0 XxNhO"
      >
        <a
          class="PrefetchLink__StyledLink-n3ecxs-0 JPiJu"
          href="/teamet/cvc"
        >
          <div
            style="height: 226px;"
            class="lazyload-placeholder"
          />
        </a>
        <div
          class="TeamMemberDuo__Meta-s9kjazo-1 gzvJxw"
        >
          <div
            class="TeamMemberDuo__MetaSection-s9kjazo-3 XeOZD"
          >
            <div
              class="TeamMemberDuo__Title-s9kjazo-2 dFgPDl"
            >
              1 ahahah
            </div>
            <div>
              Göteborg
            </div>
          </div>
          <div
            class="TeamMemberDuo__Title-s9kjazo-2 dFgPDl"
          >
            Count van Count
          </div>
          <div>
            <a
              class="s1qo6rwq-0-NativeLink-jXoiCT dJdJnO"
              href="tel:0707654321"
            >
              070-765 43 21
            </a>
          </div>
          <a
            class="NativeLink-s1qo6rwq-0 jNinQt"
            href="mailto:count@sesame.com"
          >
            count@sesame.com
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
